debezium-core/src/main/java/io/debezium/transforms/debeziumtimestampcoverter/DebeziumTimestampConverter.java

```diff
@@ -0,0 +1,698 @@
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_INT16, () -> {
+            return Schema.OPTIONAL_INT16_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_INT8, () -> {
+            return Schema.OPTIONAL_INT8_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_FLOAT64, () -> {
+            return Schema.OPTIONAL_FLOAT64_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_FLOAT32, () -> {
+            return Schema.OPTIONAL_FLOAT32_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_BOOLEAN, () -> {
+            return Schema.OPTIONAL_BOOLEAN_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(STRING_TYPE, () -> {
+            return Schema.OPTIONAL_STRING_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_BYTES, () -> {
+            return Schema.OPTIONAL_BYTES_SCHEMA.schema();
+        });
+
+        OPTIONAL_TRANSLATORS.put(TYPE_STRUCT, () -> {
+            return SchemaBuilder.type(Schema.Type.STRUCT).optional().build().schema();
+        });
+
+    }
+
+
+    public static Boolean containsField(String[] field_type, String field_name) {
+        return Stream.of(field_type).anyMatch(f -> f.equals(field_name));
+    }
+
+    // This is a bit unusual, but allows the transformation config to be passed to static anonymous classes to customize
+    // their behavior
+    public static class Config {
+        Config(String field, String type, SimpleDateFormat format, SimpleDateFormat dateFormat, String[] field_type, String[] struct_field) {
+            this.field = field;
+            this.type = type;
+            this.format = format;
+            this.dateFormat = dateFormat;
+            this.field_type = field_type;
+            this.struct_field = struct_field;
+        }
+
+        String field;
+        String type;
+        SimpleDateFormat format;
+        SimpleDateFormat dateFormat;
+        String[] field_type;
+        String[] struct_field;
+    }
+
+    private Config config;
+    private Cache<Schema, Schema> schemaUpdateCache;
+    private static final Map<String, String> TypeConverters = new HashMap<>();
+
+    @Override
+    public void configure(Map<String, ?> configs) {
+        final SimpleConfig simpleConfig = new SimpleConfig(CONFIG_DEF, configs);
+        final String field = simpleConfig.getString(FIELD_CONFIG);
+        final String type = simpleConfig.getString(TARGET_TYPE_CONFIG);
+        final String[] field_type = simpleConfig.getString(FIELD_TYPE_CONFIG).split(",");
+        final String[] struct_field = simpleConfig.getString(STRUCT_FIELD_NAME_CONFIG).split(",");
+        String[] structTypeList = new String[]{type};
+        String formatPatternTimestamp = simpleConfig.getString(TIMESTAMP_FORMAT_CONFIG);
+        String formatPatternDate = simpleConfig.getString(DATE_FORMAT_CONFIG);
+        schemaUpdateCache = new SynchronizedCache<>(new LRUCache<Schema, Schema>(16));
```

Comment:
I'd suggest adding a method to `io.debezium.util.Collect`:

```
    /**
     * Create a LRU cache.
     * @param maxSize the maximum number of entries.
     * @param <K> the type of keys
     * @param <V> the type of values
     * @return the LRU cache; never null
     */
    public static <K,V> Map<K,V> createLruCache(final int maxSize) {
        return new LinkedHashMap<K, V>(maxSize) {
            @Override
            protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
                return size() >= maxSize;
            }
        };
    }
```

so you can then use it here:

```
schemaUpdateCache = Collect.createLruCache(16);
```